### PR TITLE
[3967] Deprecate `DefaultTypingUpdatesBuffer`

### LINKED ISSUE #3967: Internalize `DefaultTypingUpdatesBuffer`
### Context & proposal

As per our slack discussion, we will internalize `DefaultTypingUpdatesBuffer`.

### Acceptance criteria

To accept this task we need to complete the following items:

- [x] Make `DefaultTypingUpdatesBuffer` internal

### PR DESCRIPTION
**Pull request by MarinTolic**
### 🎯 Goal

Closes #3967

### 🛠 Implementation details

- Move `CoroutineScope` to the constructor
- Update the documentation
- Start the deprecation process

#### Docs:

**Please run a docusaurus build and check the documentation changes**

#### Code:

Although the actual changes to the code were very slight and there is no real change in behavior expected, if you want to be thorough you can do the following on both apps.

Pre testing scenario:

1. Launch the app on two devices with two different users logged in
2. Enter the same channel

Test scenario 1:

1. On user device A start typing
2. Observe on User B that the typing update has appeared and should disappear ~3 seconds since the last keystroke on device B

Test scenario 2: 

1. Repeat the same but exist channel mid typing (the typing updates should stop)

#### General
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

gif

### CODE REVIEW

File: stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/utils/typing/DefaultTypingUpdatesBuffer.kt [hunk 1]
```diff
@@ -35,22 +35,24 @@ import kotlinx.coroutines.launch
...
+ * @param coroutineScope The coroutine scope used for running the timer and sending updates.
  * @param onTypingStarted Signals that a typing event should be sent.
  * Usually used to make an API call using [io.getstream.chat.android.client.ChatClient.keystroke]
  * @param onTypingStopped Signals that a stop typing event should be sent.
  * Usually used to make an API call using [io.getstream.chat.android.client.ChatClient.stopTyping]
  */
+@Deprecated(
+    message = "This class has been deprecated and will be removed." +
+        "Should you wish to implement your own buffer, you can create " +
+        "a custom implementation of `TypingUpdatesBuffer`.",
```
Comment (bychkovdmitry) [inline stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/utils/typing/DefaultTypingUpdatesBuffer.kt:47]: Don't think that markdown is supported here

**Review by bychkovdmitry — APPROVED** [review]
LGTM

**Comment by bychkovdmitry** [comment]
Test scenarios work as described

File: stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/utils/typing/DefaultTypingUpdatesBuffer.kt (hunk 1 above)
Comment (MarinTolic) [reply stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/utils/typing/DefaultTypingUpdatesBuffer.kt:47]: AFAIK it isn't, but I use it to try to signify that it's a piece of code in the AS info bubble.

**Pull request merged**